src/AddressLabel.jsx

```diff
@@ -0,0 +1,78 @@
+const React = require('react');
+const { ReactiveComponent } = require('oo7-react');
+const { Label, Button } = require('semantic-ui-react');
+
+class AddressLabel extends ReactiveComponent {
+	constructor () {
+		super(['address']);
+		this.state = {
+			isCopyHovered: false
+		};
+	}
+
+	onCopyAddress (text) {
+		var textArea = document.createElement('textarea');
+
+		textArea.style.position = 'fixed';
+		textArea.style.top = '-1000px';
+		textArea.style.left = '-1000px';
+		// Ensure it has a small width and height. Setting to 1px / 1em
+		// doesn't work as this gives a negative w/h on some browsers.
+		textArea.style.width = '2em';
+		textArea.style.height = '2em';
+		// We don't need padding, reducing the size if it does flash render.
+		textArea.style.padding = 0;
+		// Clean up any borders.
+		textArea.style.border = 'none';
+		textArea.style.outline = 'none';
+		textArea.style.boxShadow = 'none';
+		// Avoid flash of white box if rendered for any reason.
+		textArea.style.background = 'transparent';
+
+		textArea.value = text;
+
+		document.body.appendChild(textArea);
+
+		textArea.select();
+
+		try {
+			var successful = document.execCommand('copy');
+			var msg = successful ? 'successful' : 'unsuccessful';
+		} catch (err) {
+			console.error('unable to copy ', err);
+		}
+
+		document.body.removeChild(textArea);
+	}
+
+	handleCopyHover () {
+		this.setState({
+			isCopyHovered: !this.state.isCopyHovered
+		});
+	}
+
+	render () {
+		const { address, isHashHovered, isCopyHovered } = this.state;
+
+		if (typeof this.state.address === 'undefined') { return (<div />); }
+		return (
+			<div>
+				<Button.Group>
+					<Button
+						basic={ !isCopyHovered }
+						icon='clone'
+						color='blue'
+						onClick={ this.onCopyAddress.bind(this, address) }
+						onMouseEnter={ this.handleCopyHover.bind(this) }
```

Comment:
Small hack: Define `onCopyAddress (text) {}` as `onCopyAddress = (text) => {}` and you don't need to do .bind()

Reply:
I believe this requires making changes to babel: https://babeljs.io/docs/plugins/transform-class-properties/ not sure, do you think its worth adding the dep :)? For now I've moved the binding to constructor